odl/tomo/geometry/parallel.py

```diff
@@ -125,84 +171,222 @@ def det_to_src(self, angles, dpar, normalized=True):
             raise ValueError('`angles` {} not in the valid range {}'
                              ''.format(angles, self.motion_params))
-
         if dpar not in self.det_params:
             raise ValueError('`dpar` {} not in the valid range '
                              '{}'.format(dpar, self.det_params))
-
         if not normalized:
             raise NotImplementedError('non-normalized detector to source is '
                                       'not available in parallel case')
 
         return self.rotation_matrix(angles).dot(self.detector.normal)
 
 
-class Parallel2dGeometry(ParallelGeometry):
+class Parallel2dGeometry(ParallelBeamGeometry):
 
     """Parallel beam geometry in 2d.
 
     The motion parameter is the counter-clockwise rotation angle around
-    the origin, and the detector is a line detector perpendicular to the
-    ray direction.
+    the origin, and the detector is a line detector.
 
-    In the standard configuration, the detector reference point starts
-    at ``(1, 0)``, and the initial detector axis is ``(0, 1)``.
+    In the standard configuration, the detector is perpendicular to the
+    ray direction, its reference point is initially at ``(0, 1)``, and
+    the initial detector axis is ``(1, 0)``.
+
+    For details, check `the online docs
+    <https://odlgroup.github.io/odl/guide/geometry_guide.html>`_.
     """
 
-    def __init__(self, apart, dpart, **kwargs):
+    _default_config = dict(det_pos_init=(0, 1), det_axis_init=(1, 0))
+
+    def __init__(self, apart, dpart, det_pos_init=(0, 1), **kwargs):
         """Initialize a new instance.
 
         Parameters
         ----------
         apart : 1-dim. `RectPartition`
-            Partition of the angle interval
+            Partition of the angle interval.
         dpart : 1-dim. `RectPartition`
-            Partition of the detector parameter interval
-        det_init_pos : `array-like`, shape ``(2,)``, optional
-            Initial position of the detector reference point. The zero
-            vector is only allowed if ``det_init_axis`` is explicitly
-            given.
-            Default: ``(1, 0)``.
-        det_init_axis : `array-like` (shape ``(2,)``), optional
-            Initial axis defining the detector orientation.
-            By default, a normalized `perpendicular_vector` to
-            ``det_init_pos`` is used, which is only valid if
-            ``det_init_axis`` is not zero.
+            Partition of the detector parameter interval.
+        det_pos_init : `array-like`, shape ``(2,)``, optional
+            Initial position of the detector reference point.
+            Default: ``(0, 1)``.
+        det_axis_init : `array-like` (shape ``(2,)``), optional
+            Initial axis defining the detector orientation. The default
+            depends on ``det_pos_init``, see Notes.
+
+        Other Parameters
+        ----------------
+        init_matrix : `array_like`, shape ``(2, 2)``, optional
```

Comment:
You're right, the initializer offers two ways to do the same thing. 

My suggestion would then be to move the matrix thing to something else. Not sure however if it should be a free function. It's still a quite low-level interaction with the class, in no way simpler than the default constructor.

An alternative approach would be to make this an alternative constructor as classmethod, i.e., something like 
```python
class MyGeometry(...):
    def __init__(self, ...):
        # default constructor, no matrix

    @classmethod
    def frommatrix(cls, ..., matrix):
        # Compute vectors from defaults and `matrix`
        # and construct the instance
```